File: Récupération 20210215/ReStartAffinageAuto/Form1.vb
Public Class Form1

    Private Declare Function WritePrivateProfileString Lib "kernel32" Alias "WritePrivateProfileStringA" (ByVal MyApplicationName As String, ByVal MyKeyName As String, ByVal MykeyString As String, ByVal MyFileName As String) As Short
    Private Declare Function GetPrivateProfileString Lib "kernel32" Alias "GetPrivateProfileStringA" (ByVal NomModule As String, ByVal MotCle As String, ByVal StringParDefaut As String, ByVal ReturnedString As String, ByVal Longueur As Short, ByVal NomFichierIni As String) As Short

    ReadOnly TargetConso As String = "\\parappp01112\ARTRADIO\CONSOLIDATION\"
    ReadOnly TargetAffinageAuto As String = "\\PARAPPP01112\ARTRADIO\AFFINAGEAUTO\"

    ReadOnly szFileForceReStart As String = TargetAffinageAuto + "ConsoAffinageAuto.rst"
    ReadOnly szUserFile As String = TargetConso + "user.txt"
    ReadOnly szBaseConso As String = TargetConso + "KeyConso.db"

    Private Const MAX_LEN As Short = 640

    Public Sub Select_WriteIniString(ByVal NomModule As String, ByVal MotClé As String, ByVal Chaine As String, ByVal FichierIni As String)

        Dim Status As Short
        Status = WritePrivateProfileString(NomModule, MotClé, Chaine, FichierIni)

    End Sub

    Public Function Select_GetIniString(ByRef NomModule As String, ByRef MotCle As String, ByRef FichierIni As String) As String

        Dim Chaine As String
        Dim Status As Short

        Chaine = Space(MAX_LEN)
        Status = GetPrivateProfileString(NomModule, MotCle, "", Chaine, MAX_LEN, FichierIni)
        Select_GetIniString = Mid(Chaine, 1, Len(Trim(Chaine)) - 1)

    End Function


    Private Sub Form1_Load(sender As System.Object, e As System.EventArgs) Handles MyBase.Load

        Refresh_status()

    End Sub


    Private Sub Start_Click(sender As System.Object, e As System.EventArgs) Handles Start.Click

        If IO.File.Exists(szFileForceReStart) Then

            Refresh_status()

        ElseIf MsgBox("Etes-vous sûr de vouloir lancer le traitement ?", MsgBoxStyle.Exclamation Or MsgBoxStyle.YesNo Or MsgBoxStyle.DefaultButton2, "Status Affinage Auto") = MsgBoxResult.Yes Then

            IO.File.WriteAllText(szFileForceReStart, My.User.Name, System.Text.Encoding.UTF8)

            Refresh_status()

        End If

    End Sub

    Private Sub Rafraichir_Click(sender As System.Object, e As System.EventArgs) Handles Rafraichir.Click

        Refresh_status()

    End Sub

    Private Sub Quitter_Click(sender As System.Object, e As System.EventArgs) Handles Quitter.Click

        Me.Close()
        End

    End Sub

    Private Sub Refresh_status()

        Me.Show()

        Start.Enabled = False
        Rafraichir.Enabled = False

        If IO.File.Exists(szFileForceReStart) Then

            LabelValueStatus.Text = "En attente de traitement..."

            System.Windows.Forms.Application.DoEvents()

            While IO.File.Exists(szFileForceReStart)

                System.Windows.Forms.Application.DoEvents()

                Threading.Thread.Sleep(500)

            End While

            'Attente de creation du fichier
            While Not IO.File.Exists(szUserFile)

                System.Windows.Forms.Application.DoEvents()

                Threading.Thread.Sleep(500)

            End While

        End If

        If IO.File.Exists(szUserFile) Then

            LabelValueUser.Text = IO.File.ReadAllLines(szUserFile, System.Text.Encoding.UTF8).First
            LabelValueDate.Text = IO.File.GetLastWriteTime(szUserFile).ToString
            LabelValueStatus.Text = "En cours de traitement..."


            While IO.File.Exists(szUserFile)

                Try
                    Dim dDate As Date = IO.File.GetLastWriteTime(szUserFile)

                    Dim svValue As String = Conversion(DateDiff(DateInterval.Second, dDate, DateTime.Now))
                    If String.Compare(Rafraichir.Text, svValue) <> 0 Then
                        Rafraichir.Text = svValue
                    End If

                Catch ex As Exception

                End Try

                System.Windows.Forms.Application.DoEvents()

                Threading.Thread.Sleep(100)

            End While

            Rafraichir.Text = "Rafraichir"

        End If

        'attente de copie de la base

        LabelValueUser.Text = "Aucun"
        LabelValueDate.Text = IO.File.GetLastWriteTime(szBaseConso).ToString
        LabelValueStatus.Text = "traitement terminé."
        Start.Enabled = True
        Rafraichir.Enabled = True

    End Sub


    Private Function Conversion(ByVal TempsTotal As Long) As String
        If TempsTotal > 86400 Then Return "00:00:00"
        Dim Heures As Integer = Int(TempsTotal / 3600)
        Dim Minutes As Integer = Int((TempsTotal - 3600 * Heures) / 60)
        Dim Secondes As Integer = TempsTotal - 3600 * Heures - 60 * Minutes
        Return Format(Heures, "##00") + ":" + Format(Minutes, "##00") + ":" + Format(Secondes, "##00")
    End Function

End Class

File: Kantar.MediaPlanning.AffinageAutoStartTool/Kantar.MediaPlanning.AffinageAutoStartTool/Form1.cs
using Kantar.MediaPlanning.Logger;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Configuration;
using System.Data;
using System.Drawing;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace Kantar.MediaPlanning.AffinageAutoStartTool
{
    public partial class Form1 : Form
    {

        #region Global members
        public string ConsoPath { get; set; }
        public string AffinageAutoPath { get; set; }
        public string FileForceReStart { get; set; }
        public string UserFilePath { get; set; }
        public string BaseConsoFilePath { get; set; }

        #endregion

        #region Constructor
        public Form1()
        {
            InitializeComponent();
			this.Icon = Properties.Resources.logokantar;
			loadConfigfile();
		}
        #endregion

        #region Private Methods
        private void refresh_status()
		{
			TraceLoggingtool.WriteLineIf(System.Diagnostics.TraceLevel.Verbose, "refresh_status -> start");
			Show();
			this.buttonStart.Enabled = false;
			this.buttonRefresh.Enabled = false;
			if (File.Exists(FileForceReStart))
			{
				TraceLoggingtool.WriteLineIf(System.Diagnostics.TraceLevel.Verbose, "refresh_status -> file exists :"+ FileForceReStart);
				this.labelStatusValue.Text = "En attente de traitement...";
				Application.DoEvents();
				while (File.Exists(FileForceReStart))
				{
					Application.DoEvents();
					Thread.Sleep(500);
				}
				while (!File.Exists(UserFilePath))
				{
					Application.DoEvents();
					Thread.Sleep(500);
				}
			}
			else
            {
				TraceLoggingtool.WriteLineIf(System.Diagnostics.TraceLevel.Verbose, "refresh_status -> file not exists :" + FileForceReStart);
			}
			if (File.Exists(UserFilePath))
			{
				TraceLoggingtool.WriteLineIf(System.Diagnostics.TraceLevel.Verbose, "refresh_status -> file exist :" + UserFilePath);
				try
				{
					this.labelValueUser.Text = File.ReadAllLines(UserFilePath, Encoding.UTF8).First();
				}
				catch(Exception ex)
                {
					TraceLoggingtool.WriteLineIf(System.Diagnostics.TraceLevel.Error, "refresh_status -> ReadAllLines error :" + ex.Message);
				}
				this.labelValueDate.Text = File.GetLastWriteTime(UserFilePath).ToString();
				this.labelStatusValue.Text = "En cours de traitement...";
				while (File.Exists(UserFilePath))
				{
					try
					{
						DateTime lastWriteTime = File.GetLastWriteTime(UserFilePath);
						//string strB = Conversion(DateTime.DateDiff(DateInterval.Second, lastWriteTime, DateTime.Now));
						var diff = ((TimeSpan)(DateTime.Now - lastWriteTime));
						//string strB = Conversion(diff);
						if (string.Compare(this.buttonRefresh.Text, diff.ToString(@"hh\:mm\:ss")) != 0)
						{
							this.buttonRefresh.Text = diff.ToString(@"hh\:mm\:ss");
						}
					}
					catch (Exception ex)
					{
						TraceLoggingtool.WriteLineIf(System.Diagnostics.TraceLevel.Error, "refresh_status error -> lastWriteTime" + ex.Message);
					}
					Application.DoEvents();
					Thread.Sleep(100);
				}
				this.buttonRefresh.Text = "Rafraichir";
			}
			else
			{
				TraceLoggingtool.WriteLineIf(System.Diagnostics.TraceLevel.Verbose, "refresh_status -> file not exists :" + UserFilePath);
			}
			this.labelValueUser.Text = "Aucun";
			try
			{
				this.labelValueDate.Text = File.GetLastWriteTime(BaseConsoFilePath).ToString();
				this.labelStatusValue.Text = "Traitement terminé.";
			}
			catch(Exception ex)
            {
				TraceLoggingtool.WriteLineIf(System.Diagnostics.TraceLevel.Error, "BaseConsoFilePath :" + BaseConsoFilePath);
				TraceLoggingtool.WriteLineIf(System.Diagnostics.TraceLevel.Error, "BaseConsoFilePath :" + ex.Message);
				this.labelStatusValue.Text = "Erreur.";
			}
			this.buttonStart.Enabled = true;
			this.buttonRefresh.Enabled = true;
		}
		private string Conversion(double TempsTotal)
		{
			TraceLoggingtool.WriteLineIf(System.Diagnostics.TraceLevel.Error, "Conversion :" + TempsTotal);
			if (TempsTotal > 86400)
			{
				return "00:00:00";
			}
			else
			{
				TimeSpan time = TimeSpan.FromSeconds(TempsTotal);
				return time.ToString(@"hh\:mm\:ss");
			}
		}
		private void loadConfigfile()
		{
			TraceLoggingtool.WriteLineIf(System.Diagnostics.TraceLevel.Verbose, "LoadConfigfile -> start");
			try
			{
				ConsoPath = ConfigurationManager.AppSettings["TargetConsoPath"];
				AffinageAutoPath = ConfigurationManager.AppSettings["TargetAffinageAutoPath"];
				UserFilePath = ConfigurationManager.AppSettings["UserFile"];
				BaseConsoFilePath = ConfigurationManager.AppSettings["BaseConsoFile"];
				FileForceReStart = ConfigurationManager.AppSettings["FileForceReStart"];
				FileForceReStart = Path.Combine(AffinageAutoPath, FileForceReStart);
				UserFilePath = Path.Combine(ConsoPath, UserFilePath);
				BaseConsoFilePath = Path.Combine(ConsoPath, BaseConsoFilePath);
			}
			catch (Exception ex)
			{
				TraceLoggingtool.WriteLineIf(System.Diagnostics.TraceLevel.Error, "LoadConfigfile error :" + ex.Message);
				this.labelStatus.Text = "Erreur chargement config";
			}
		}
		#endregion

		#region Events

		private void buttonRefresh_Click(object sender, EventArgs e)
        {
            refresh_status();
        }

        private void buttonStart_Click(object sender, EventArgs e)
        {
			TraceLoggingtool.WriteLineIf(System.Diagnostics.TraceLevel.Verbose, "buttonStart_Click -> start");

			if (File.Exists(FileForceReStart))
			{
				TraceLoggingtool.WriteLineIf(System.Diagnostics.TraceLevel.Verbose, "buttonStart_Click -> file exists :"+ FileForceReStart);
				refresh_status();
			}
			else if (MessageBox.Show("Etes-vous sûr de vouloir lancer le traitement ?", "Status Affinage Auto", MessageBoxButtons.YesNo, MessageBoxIcon.Exclamation) == DialogResult.Yes)
			{
				TraceLoggingtool.WriteLineIf(System.Diagnostics.TraceLevel.Verbose, "buttonStart_Click -> Msgbox Yes");
				//File.WriteAllText(FileForceReStart, MyProject.User.Name, Encoding.UTF8);
				File.WriteAllText(FileForceReStart, System.Security.Principal.WindowsIdentity.GetCurrent().Name, Encoding.UTF8);
				refresh_status();
			}
		}

        private void buttonExit_Click(object sender, EventArgs e)
        {
            Close();
        }

		private void Form1_Load(object sender, EventArgs e)
		{
			refresh_status();
		}

		#endregion


	}
}
